sql/core/src/main/scala/org/apache/spark/sql/mathfunctions.scala

```diff
@@ -0,0 +1,562 @@
+   *
+   * @group int_funcs
+   */
+  def isignum(e: Column): Column = ISignum(e.expr)
+
+  /**
+   * Computes the signum of the given column. For IntegerType.
+   *
+   * @group int_funcs
+   */
+  def isignum(columnName: String): Column = isignum(Column(columnName))
+
+  /**
+   * Computes the signum of the given value. For FloatType.
+   *
+   * @group float_funcs
+   */
+  def fsignum(e: Column): Column = FSignum(e.expr)
+
+  /**
+   * Computes the signum of the given column. For FloatType.
+   *
+   * @group float_funcs
+   */
+  def fsignum(columnName: String): Column = fsignum(Column(columnName))
+
+  /**
+   * Computes the signum of the given value. For LongType.
+   *
+   * @group long_funcs
+   */
+  def lsignum(e: Column): Column = LSignum(e.expr)
+
+  /**
+   * Computes the signum of the given column. For FloatType.
+   *
+   * @group long_funcs
+   */
+  def lsignum(columnName: String): Column = lsignum(Column(columnName))
+
+  /**
+   * Computes the natural logarithm of the given value.
+   *
+   * @group double_funcs
+   */
+  def log(e: Column): Column = Log(e.expr)
+
+  /**
+   * Computes the natural logarithm of the given column.
+   *
+   * @group double_funcs
+   */
+  def log(columnName: String): Column = log(Column(columnName))
+
+  /**
+   * Computes the logarithm of the given value in Base 10.
+   *
+   * @group double_funcs
+   */
+  def log10(e: Column): Column = Log10(e.expr)
+
+  /**
+   * Computes the logarithm of the given value in Base 10.
+   *
+   * @group double_funcs
+   */
+  def log10(columnName: String): Column = log10(Column(columnName))
+
+  /**
+   * Computes the natural logarithm of the given value plus one.
+   *
+   * @group double_funcs
+   */
+  def log1p(e: Column): Column = Log1p(e.expr)
+
+  /**
+   * Computes the natural logarithm of the given column plus one.
+   *
+   * @group double_funcs
+   */
+  def log1p(columnName: String): Column = log1p(Column(columnName))
+
+  /**
+   * Computes the exponential of the given value.
+   *
+   * @group double_funcs
+   */
+  def exp(e: Column): Column = Exp(e.expr)
+
+  /**
+   * Computes the exponential of the given column.
+   *
+   * @group double_funcs
+   */
+  def exp(columnName: String): Column = exp(Column(columnName))
+
+  /**
+   * Computes the exponential of the given value minus one.
+   *
+   * @group double_funcs
+   */
+  def expm1(e: Column): Column = Expm1(e.expr)
+
+  /**
+   * Computes the exponential of the given column.
+   *
+   * @group double_funcs
+   */
+  def expm1(columnName: String): Column = expm1(Column(columnName))
+
+  /**
+   * Returns the value of the first argument raised to the power of the second argument.
+   *
+   * @group double_funcs
+   */
+  def pow(l: Column, r: Column): Column = Pow(l.expr, r.expr)
+
+  /**
+   * Returns the value of the first argument raised to the power of the second argument.
+   *
+   * @group double_funcs
+   */
+  def pow(l: Column, rightName: String): Column = pow(l, Column(rightName))
```

Comment:
why don't we remove this for now? seems weird to have this. I think xiangrui was asking for pow(l: Column, l: Double), and pow(l: Double, r: Column)